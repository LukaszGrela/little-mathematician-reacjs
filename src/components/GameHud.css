.game-hud {
    background-color: transparent;
    padding: 10px 0;
    margin-bottom: 10px;
}
.game-hud .container {
    width: 300px;
    margin: 0 auto;
}
.game-hud .container div {
    display: inline-block;
    padding: 0 10px;
    width: 80px;
}

.question-counter {
    text-align: left;
}

.correct-answers {
    color: green;
    text-align: right;
}

.game-hud.addition {
    border-bottom: 2px solid #388E3C;
}

.game-hud.subtraction {
    border-bottom: 2px solid #8BC34A;
}

.game-hud.multiplication {
    border-bottom: 2px solid #CDDC39;
}

.game-hud.division {
    border-bottom: 2px solid #FFEB3B;
}