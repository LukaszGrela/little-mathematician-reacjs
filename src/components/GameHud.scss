/*
   Copyright 2018 ≈Åukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

@import '../sass/_variables.scss';

.game-hud {
    background-color: transparent;
    padding: 10px 0;
    margin-bottom: 10px;
    .container {
        width: $width-game-column;
        margin: 0 auto;
        div {
            display: inline-block;
            padding: 0 10px;
            width: 80px;
        }
    }
    /* bottom border of the hud */
    &.addition {
        border-bottom: 2px solid $accent-color-1;
    }
    &.subtraction {
        border-bottom: 2px solid $accent-color-2;
    }
    &.multiplication {
        border-bottom: 2px solid $accent-color-3;
    }
    &.division {
        border-bottom: 2px solid $accent-color-4;
    }
}

.question-counter {
    text-align: left;
}

.icon-grade {
    width: 24px;
    height: 24px;
    position: relative;
    top: 5px;
    margin-right: 10px
}

.correct-answers {
    color: $color-correct;
    text-align: right;
    svg.icon-grade {
        fill: $color-gold-star;
    }
}

.gain-even .icon-grade,
.gain-odd .icon-grade {
    animation-duration: 0.5s;
    animation-iteration-count: 3;
    animation-timing-function: ease-out;
}

.gain-even .icon-grade {
    animation-name: zoom-star-1;
}

.gain-odd .icon-grade {
    animation-name: zoom-star-2;
}

@keyframes zoom-star-1 {
    0% {
        transform: scale(1);
    }
    50% {
        opacity: .75;
    }
    100% {
        transform: scale(2);
        opacity: 0;
    }
}

@keyframes zoom-star-2 {
    0% {
        transform: scale(1);
    }
    50% {
        opacity: .75;
    }
    100% {
        transform: scale(2);
        opacity: 0;
    }
}