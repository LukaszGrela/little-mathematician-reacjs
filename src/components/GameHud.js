/*
   Copyright 2018 ≈Åukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

import React, { Component } from 'react';

import IconGrade from '../icons/IconGrade';

import './GameHud.css'

class GameHud extends Component {
    state = {
        scored: false
    }

    componentWillReceiveProps(nextProps) {

        //if (nextProps.hudCorrectAnswers !== this.props.hudCorrectAnswers) {
        //}
    }


    componentDidMount() {
        console.log('componentDidMount');
    }
    componentWillMount() {
        console.log('componentWillMount');
    }

    componentWillUnmount() {
        console.log('componentWillUnmount');
    }

    render() {
        return (
            <div className={'game-hud '
                + this.props.type.split(':').join('')
                + (this.props.hudCorrectAnswers > 0 ?
                    ' gain-' + (this.props.hudCorrectAnswers % 2 ? 'even' : 'odd')
                    :''
                )
            }>
                <div className='container'>
                    <div className='question-counter'>{this.props.hudQuestionCurrent} / {this.props.questionCount}</div>
                    <div className='correct-answers'>
                        <IconGrade className='icon-grade' />
                        <span className='label'>{this.props.hudCorrectAnswers}</span>
                    </div>
                </div>
            </div>
        );
    }
}

export default GameHud;