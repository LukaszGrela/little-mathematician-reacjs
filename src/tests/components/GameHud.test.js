import React from 'react';
import { shallow } from 'enzyme';
import GameHud from '../../components/GameHud';
import { GAME_ADDITION } from '../../gameTypes';

let wrapper, type, state;

beforeEach(() => {
    type = GAME_ADDITION;
    state = {
        hudCorrectAnswers: 0,
        hudQuestionCurrent: 1,
        questionCount: 1,
        type
    };
    wrapper = shallow(<GameHud
        {...state}
        type={type}
    />);
})
test('Should render GameHud correctly', () => {
    expect(wrapper).toMatchSnapshot();
});
test('Should render GameHud correctly with different props', () => {
    wrapper.setProps({
        hudQuestionCurrent:2,
        hudCorrectAnswers:3,
        questionCount:10
    });
    expect(wrapper).toMatchSnapshot();
});
// 
test('Should GameHud className contain prop type class', () => {
    const type = 'my-type';
    wrapper.setProps({
        type
    });
    expect(wrapper.find('.game-hud.'+type)).toHaveLength(1);
});
// 
test('Should GameHud className contain gain-even class', () => {
    let hudCorrectAnswers = 2;
    wrapper.setProps({
        hudCorrectAnswers
    });
    expect(wrapper.find('.game-hud.gain-even')).toHaveLength(1);
});
// 
test('Should GameHud className contain gain-odd class', () => {
    const hudCorrectAnswers = 3;
    wrapper.setProps({
        hudCorrectAnswers
    });
    expect(wrapper.find('.game-hud.gain-odd')).toHaveLength(1);
});
// 
test('Should GameHud display progress correctly', () => {
    const hudQuestionCurrent = 5;
    const questionCount = 10;
    wrapper.setProps({
        hudQuestionCurrent,
        questionCount,
    });
    const progress = wrapper.find('.question-counter');
    expect(progress).toHaveLength(1);

    expect(progress.text()).toBe(`${hudQuestionCurrent} / ${questionCount}`);
});
//
test('Should GameHud display correct answers number correctly', () => {
    const hudCorrectAnswers = 10;
    wrapper.setProps({
        hudCorrectAnswers
    });
    const score = wrapper.find('.correct-answers span');
    expect(score).toHaveLength(1);

    expect(score.text()).toBe(`${hudCorrectAnswers}`);
  
});