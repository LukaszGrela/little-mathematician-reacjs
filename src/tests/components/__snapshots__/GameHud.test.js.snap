// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render GameHud correctly 1`] = `
<div
  className="game-hud addition"
>
  <div
    className="container"
  >
    <div
      className="question-counter"
    >
      1
       / 
      1
    </div>
    <div
      className="correct-answers"
    >
      <IconGrade
        className="icon-grade"
      />
      <span
        className="label"
      >
        0
      </span>
    </div>
  </div>
</div>
`;

exports[`Should render GameHud correctly with different props 1`] = `
<div
  className="game-hud addition gain-odd"
>
  <div
    className="container"
  >
    <div
      className="question-counter"
    >
      2
       / 
      10
    </div>
    <div
      className="correct-answers"
    >
      <IconGrade
        className="icon-grade"
      />
      <span
        className="label"
      >
        3
      </span>
    </div>
  </div>
</div>
`;
